### PR TITLE
Add conformance test for zero fields in oneofs.

### PR DESCRIPTION
**Pull request by thomasvl**

### CODE REVIEW

File: conformance/conformance_test.cc [hunk 1]
```diff
@@ -1225,6 +1241,34 @@ bool ConformanceTestSuite::RunSuite(ConformanceTestRunner* runner,
   ExpectParseFailureForJson(
       "OneofFieldDuplicate",
       R"({"oneofUint32": 1, "oneofString": "test"})");
+  // Ensure zero values for oneof make it out/backs.
+  {
+    TestAllTypes message;
+    message.set_oneof_uint32(0);
+    RunValidProtobufTest(
+        "Oneof_Zero_uint32", message, "oneof_uint32: 0");
```
Comment (xfxyjwf) [inline conformance/conformance_test.cc]: Can you name the test case consistently with others?

Comment (thomasvl) [inline conformance/conformance_test.cc]: Sure, what format would you like?  Just drop the underscores?  OneofZeroUint32, etc.?

Comment (xfxyjwf) [inline conformance/conformance_test.cc]: OneofZeroUint32 seems good

Comment (thomasvl) [inline conformance/conformance_test.cc]: Done.

**Comment by thomasvl** [comment]
Letting presubmits run again, and then I'll land as a squashed single commit.
